contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -560,8 +560,8 @@ impl InterchainTokenService {
         amount: i128,
     ) {
         // Due to limitations of the soroban-sdk, there is no type-safe client for contract execution.
-        // The invocation will panic on error, so we can safely cast the return value to `()` and discard it.
-        env.invoke_contract::<()>(
+        // The invocation might return a value, so we use Val as the return type to avoid panics
+        env.invoke_contract::<Val>(
```

Comment:
check other uses of `invoke_contract` in the repo to see if it's applicable elsewhere